### ISSUE TITLE
Is distributed mpirun possible ?

### ISSUE THREAD

**Issue by kuizhiqing** [open-mpi/ompi#9346]
As I know, mpi application can be launch by execute following command ONLY on node1

```shell
mpirun --hostfile my_hostfile -np 2 hello_world_mpi
```
with
```shell
$ cat my_hostfile
node1.example.com slots=1
node2.example.com slots=1
```
then *hello_world_mpi* will be excuted on both node1 and node2 even without ensuring *hello_world_mpi* exists on node2.

Let's call it imperative mode.

I was wondering should/how we support the following scenarios,

### proposition 1
run 
```shell
hello_world_mpi --hostfile my_hostfile
```
on both node1 and node2, which will wait each other until ready before execution, application *hello_world_mpi* may be rewritten with ompi api.

### proposition 2
run 
```shell
mpirun --hostfile my_hostfile -np 2 hello_world_mpi
```
on both node1 and node2, which will wait each other until ready before execution.

The difference between this one with imperative mode is that command will be executed in both node manually which did not depend on rsh/ssh.

Is it possible to support or any argument or technical consideration about this ?

Thanks in advance for your reply.

**Comment by jsquyres** [comment]
Open MPI does not currently support this model of execution in any existing release.  The upcoming 5.0 won't support it, either.

That being said, the upcoming Open MPI v5.0 will be using PRRTE for its underlying run-time system (the ORTE system in all versions up to Open MPI v4.x basically split off and became its own project: https://github.com/openpmix/prrte).  It might be useful to discuss such an execution model over in the PRRTE community to see if there is any interest / desire over there.

**Comment by rhc54** [comment]
Yeah, you can do it with PMIx and PRRTE - I use it to do a population simulation where everything starts as a singleton. You must have PRRTE running in advance of starting the procs, and your MPI must be PMIx-aware (ala OMPI).
